### PR TITLE
Task 3

### PR DESCRIPTION
**Pull request by Dzmitry-Chervanenka**
- Product Service Serverless config contains configuration for 2 lambda functions, API is not working at all, but YAML configuration is correct ✅ 
 - The getProductsList OR getProductsById lambda function returns a correct response (POINT1) ✅ 
- The getProductsById AND getProductsList lambda functions return a correct response code (POINT2) ✅ 
 - Your own Frontend application is integrated with Product Service (/products API) and products from Product Service are represented on Frontend. AND POINT1 and POINT2 are done. ✅ 
 
OPTIONAL TASKS:
+5 - Async/await is used in lambda functions 🙅‍♂️ 
+5 - ES6 modules are used for Product Service implementation🙅‍♂️ 
+4 - Custom Webpack/ESBuild/etc is manually configured for Product Service. Not applicable for preconfigured/built-in bundlers that come with templates, plugins, etc. 🙅‍♂️ 
+4 (All languages) - SWAGGER documentation is created for Product Service🙅‍♂️ 
+4 (All languages) - Lambda handlers are covered by basic UNIT tests (NO infrastructure logic is needed to be covered)🙅‍♂️ 
+4 (All languages) - Lambda handlers (getProductsList, getProductsById) code is written not in 1 single module (file) and separated in codebase. 🙅‍♂️ 
+4 (All languages) - Main error scenarios are handled by API ("Product not found" error). ✅ 

[FE pull request](https://github.com/Dzmitry-Chervanenka/shop-react-redux-cloudfront/pull/2)
[Deployed FE](https://d1woy1337wrj09.cloudfront.net/)

### CODE REVIEW

File: product-service/src/functions/getProductsById/handler.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import type { ValidatedEventAPIGatewayProxyEvent } from '@libs/api-gateway';
+import { formatJSONResponse } from '@libs/api-gateway';
+import { middyfy } from '@libs/lambda';
+import products from "@functions/mock-db";
+
+
+const getProductsList: ValidatedEventAPIGatewayProxyEvent<null> = async ({pathParameters: {productId}}) => {
```
Comment (razumovich) [inline product-service/src/functions/getProductsById/handler.ts:7]: Probably, it should be renamed to getProductById

File: product-service/src/functions/hello/handler.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import type { ValidatedEventAPIGatewayProxyEvent } from '@libs/api-gateway';
```
Comment (razumovich) [inline product-service/src/functions/hello/handler.ts:1]: Was "hello" function committed by mistake? Guess, we don't have this requirement in Task 3

File: product-service/src/functions/getProductsList/index.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import { handlerPath } from '@libs/handler-resolver';
+
+export default {
+    handler: `${handlerPath(__dirname)}/handler.main`,
+    events: [
+        {
+            http: {
+                method: 'get',
+                path: 'product/available',
```
Comment (razumovich) [inline product-service/src/functions/getProductsList/index.ts:9]: Not a big deal, but I think it's better to keep up with the best practise of REST and create URLs like:

/products/{productId} - for a product
/products - for the list of products

**Review by razumovich — APPROVED** [review]

File: product-service/src/functions/getProductsById/handler.ts (hunk 1 above)
Comment (Dzmitry-Chervanenka) [reply product-service/src/functions/getProductsById/handler.ts:7]: fixed in pr for Task 4

File: product-service/src/functions/hello/handler.ts (hunk 2 above)
Comment (Dzmitry-Chervanenka) [reply product-service/src/functions/hello/handler.ts:1]: fixed in pr for Task 4

File: product-service/src/functions/getProductsList/index.ts (hunk 3 above)
Comment (Dzmitry-Chervanenka) [reply product-service/src/functions/getProductsList/index.ts:9]: fixed in pr for Task 4

**Pull request merged**